build.zig

```diff
@@ -51,34 +43,16 @@ pub fn build(b: *std.Build) void {
     const test_step = b.step("test", "Run unit tests");
     test_step.dependOn(&run_unit_tests.step);
 
-    // docs
-    {
-        const install_docs = b.addInstallDirectory(.{
-            .source_dir = lib.getEmittedDocs(),
-            .install_dir = .prefix,
-            .install_subdir = "docs",
-        });
-        const docs_step = b.step("docs", "Generate documentation");
-        docs_step.dependOn(&install_docs.step);
-    }
+    const install_docs = b.addInstallDirectory(.{
+        .source_dir = lib.getEmittedDocs(),
+        .install_dir = .prefix,
+        .install_subdir = "docs",
+    });
+    const docs_step = b.step("docs", "Generate documentation");
+    docs_step.dependOn(&install_docs.step);
 
     // check
-    {
-        const lib_check = b.addStaticLibrary(.{
-            .name = "smart-pointers",
-            // In this case the main source file is merely a path, however, in more
-            // complicated build scripts, this could be a generated file.
-            .root_source_file = b.path("src/root.zig"),
-            .target = target,
-            .optimize = optimize,
-        });
-        const test_check = b.addTest(.{
-            .root_source_file = b.path("src/root.zig"),
-            .target = target,
-            .optimize = optimize,
-        });
-        const check = b.step("check", "Check for semantic issues");
-        check.dependOn(&lib_check.step);
-        check.dependOn(&test_check.step);
-    }
+    const check = b.step("check", "Check for semantic issues");
+    check.dependOn(b.getInstallStep()); // build library
+    check.dependOn(&run_unit_tests.step); // run tests
```

Comment:
The check step shouldn't actually run tests. It should depend on `unit_rests.step`